File: benchmark-java-sync/src/main/java/com/example/benchmarkjavasync/impl/ApplicationMapper.java
package com.example.benchmarkjavasync.impl;

import com.example.benchmarkjavasync.core.model.*;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
public class ApplicationMapper {

    public Application mapToApplication(ApplicationServiceRequest application) {
        return new Application(
                UUID.randomUUID(),
                application.firstName(),
                application.lastName(),
                application.passportSerialNumber(),
                application.passportNumber(),
                application.amount(),
                application.status()
        );
    }

    public CreditInformation mapToCreditInformation(Application application) {
        return new CreditInformation(
                UUID.randomUUID(),
                application.status(),
                application.amount(),
                application.userId()
        );
    }

    public User mapToUser(Application application) {
        return new User(
                application.userId(),
                application.firstName(),
                application.lastName()
        );
    }

    public Passport mapToPassport(Application application) {
        return new Passport(
                UUID.randomUUID(),
                application.passportSerialNumber(),
                application.passportNumber(),
                application.userId()
        );
    }

    public ApplicationResponse mapToResult(UUID userId) {
        return new ApplicationResponse(userId);
    }
}

File: benchmark-kotlin-async/src/main/kotlin/com/example/benchmarkkotlinasync/impl/ApplicationMapper.kt
package com.example.benchmarkkotlinasync.impl;

import com.example.benchmarkkotlinasync.core.model.*
import org.springframework.stereotype.Service
import java.util.*

@Service
class ApplicationMapper {

    suspend fun mapToApplication(application: ApplicationServiceRequest): Application {
        return Application(
                UUID.randomUUID(),
                application.firstName,
                application.lastName,
                application.passportSerialNumber,
                application.passportNumber,
                application.amount,
                application.status)
    }

    suspend fun mapToCreditInformation(application: Application): CreditInformation {
        return CreditInformation(
                UUID.randomUUID(),
                application.status,
                application.amount,
                application.userId
        )
    }

    suspend fun mapToUser(application: Application): User {
        return User(
                application.userId,
                application.firstName,
                application.lastName
        )
    }

    suspend fun mapToPassport(application: Application): Passport {
        return Passport(
                UUID.randomUUID(),
                application.passportSerialNumber,
                application.passportNumber,
                application.userId
        )
    }

    suspend fun mapToResult(userId: UUID): ApplicationResponse {
        return ApplicationResponse(userId)
    }
}
